# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditAggregateTransaction.cpp
# Commit: b9ebe58001d7
# Full Hash: b9ebe58001d787bb0bbad6d39ae8681966d5a77f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-04 09:29:13
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r ce2087ddab5c -r b9ebe58001d7 editor/libeditor/EditAggregateTransaction.cpp
--- a/editor/libeditor/EditAggregateTransaction.cpp	Fri Apr 03 08:30:37 2020 +0000
+++ b/editor/libeditor/EditAggregateTransaction.cpp	Fri Apr 03 08:32:01 2020 +0000
@@ -24,8 +24,8 @@
 NS_IMETHODIMP EditAggregateTransaction::DoTransaction() {
   // FYI: It's legal (but not very useful) to have an empty child list.
   AutoTArray<OwningNonNull<EditTransactionBase>, 10> children(mChildren);
-  for (auto& childTransaction : children) {
-    nsresult rv = childTransaction->DoTransaction();
+  for (OwningNonNull<EditTransactionBase>& childTransaction : children) {
+    nsresult rv = MOZ_KnownLive(childTransaction)->DoTransaction();
     if (NS_FAILED(rv)) {
       NS_WARNING("EditTransactionBase::DoTransaction() failed");
       return rv;
@@ -38,8 +38,9 @@
   // FYI: It's legal (but not very useful) to have an empty child list.
   // Undo goes through children backwards.
   AutoTArray<OwningNonNull<EditTransactionBase>, 10> children(mChildren);
-  for (auto& childTransaction : Reversed(children)) {
-    nsresult rv = childTransaction->UndoTransaction();
+  for (OwningNonNull<EditTransactionBase>& childTransaction :
+       Reversed(children)) {
+    nsresult rv = MOZ_KnownLive(childTransaction)->UndoTransaction();
     if (NS_FAILED(rv)) {
       NS_WARNING("EditTransactionBase::UndoTransaction() failed");
       return rv;
@@ -51,8 +52,8 @@
 NS_IMETHODIMP EditAggregateTransaction::RedoTransaction() {
   // It's legal (but not very useful) to have an empty child list.
   AutoTArray<OwningNonNull<EditTransactionBase>, 10> children(mChildren);
-  for (auto& childTransaction : children) {
-    nsresult rv = childTransaction->RedoTransaction();
+  for (OwningNonNull<EditTransactionBase>& childTransaction : children) {
+    nsresult rv = MOZ_KnownLive(childTransaction)->RedoTransaction();
     if (NS_FAILED(rv)) {
       NS_WARNING("EditTransactionBase::RedoTransaction() failed");
       return rv;